### ISSUE TITLE
Dante internal error

### ISSUE THREAD

**Issue by jaid33** [concision/docker-windscribe-socks-server#5]
Hi,

I get this error since a couple of days (not sure if linked to any update). It was working for >6 month.
I tried to create a new container with different options, but always the same issue.
The container continously tries to restart and fails.

1970-02-28 01:17:47 OUT [SOCKS5] Searching for SOCKS_USERNAME and SOCKS_PASSWORD environment variables
1970-02-19 14:46:19 OUT [SOCKS5] No non-empty SOCKS5 accounts were received (check that the environment variables are set correctly)
1970-02-21 22:31:23 OUT [OPENVPN] Creating OpenVPN TUN device
1970-02-21 15:41:47 OUT [RESOLV] Writing /etc/resolv.conf
1970-02-21 02:02:35 OUT [RESOLV] nameserver 1.1.1.1
1970-02-23 20:02:03 OUT [WINDSCRIBE] Starting Windscribe client
1970-02-25 02:45:15 OUT [WINDSCRIBE] Starting windscribe ... OK
1970-02-26 14:01:31 OUT [WINDSCRIBE] Authenticating to Windscribe
1970-02-28 04:42:35 ERR [WINDSCRIBE] /tmp/_MEIdnUwdx/getpass.py:83: GetPassWarning: Can not control echo on the terminal.
1970-02-28 18:21:47 ERR [WINDSCRIBE] Warning: Password input may be echoed.
1970-02-28 03:34:19 ERR [WINDSCRIBE] Windscribe Password:
1970-02-19 22:44:11 OUT [WINDSCRIBE] Windscribe Username: API Error: No API Connectivity
1970-02-22 07:37:31 OUT [WINDSCRIBE] Please login to use Windscribe
1970-02-19 04:31:55 OUT [WINDSCRIBE] Connecting to Windscribe
1970-02-19 23:52:27 OUT [WINDSCRIBE] Windscribe location: NL
1970-03-01 14:50:35 OUT [WINDSCRIBE] -Please login to use Windscribe
1970-02-24 16:30:51 OUT [WINDSCRIBE] Enabling Windscribe firewall
1970-03-01 14:50:35 OUT [WINDSCRIBE] Please login to use Windscribe
1970-02-20 04:25:31 OUT [DANTE] Generating configuration file
1970-02-19 15:54:35 OUT [DANTE] Starting Danted server
1970-02-20 07:50:19 ERR [DANTE] (0.000000) danted[216]: warning: an internal error was detected at ../lib/time.c:75, value 0, expression "rc == 0", errno 1 (Operation not permitted).  Version: $Id: time.c,v 1.9 2013/10/25 12:55:01 karls Exp $.  Please report this to Inferno Nettverk A/S at "dante-bugs@inet.no".  Please check for a coredump too.

**Comment by concision** [comment]
There was an update I pushed almost 3 weeks ago that might be causing this behavior. For the time being, you can try specifying version tag `0.2.0` (see [tags](https://hub.docker.com/r/concisions/windscribe-socks-server/tags)) to see if that works on your machine.

There seems to be several issues at play here with this log - it seems to have the UNIX time set close to 0 and no network connectivity. The dates are extremely out of order and highly inconsistent...
- Do you have any other Docker containers experiencing issues?
- Are you experiencing any system instability?
- Can you try restarting the host machine to see if that fixes it?

**Comment by jaid33** [comment]
I can't use version 0.2.0 as there is no ARM-version available (I'm running docker on a Raspberry Pi).
Did this version get lost somehow?
I tried version 0.2.1 and it is showing the same issue as 0.3.0.
A restart didn't fix it, my other three containers are working just fine.

**Comment by concision** [comment]
I checked and I do not think the version `0.2.0` tag got lost somehow - I did not build a multiarch image back then. The last update to `0.2.0` was 8 months ago. You said you have been using this image for 6 months - did you build from sources originally? It should not have worked otherwise.

There is some indication this might be a bug with either buildx or ubuntu's `latest`/`focal` tag on Docker (see [this Reddit post](https://www.reddit.com/r/docker/comments/hdru82/time_in_docker_container_returns_random_value/) and the corresponding [bug report](https://bugs.launchpad.net/cloud-images/+bug/1896443)) . I _did_ update from `groovy` to `focal` in my recent changes, which did not appear to affect `amd/64` built images.

I will be pushing a an experimental image shortly might resolve this issue for you by changing the underlying base image. If it works for you, I will push it as `0.3.1`.

**Comment by concision** [comment]
**tl;dr**: try `concisions/windscribe-socks-server:0.3.1x` and let me know if that resolves your issue; if it does, I will push this as `0.3.1` afterwards.

I broke out an old Raspberry Pi of mine to try and replicate your issue. Unfortunately, I was unable to replicate the exact issue, but did experience a similar issue with the Windscribe API connectivity issue in your logs. I changed the base image back to a `ubuntu:bionic` tag (ubuntu 18) which was used in previous versions. This resolved the connectivity issue for me. 

This aforementioned connectivity issue, in conjunction with the bug report I found, seems to indicate there is an incompatibility with `linux/arm/v7` and `ubuntu:focal` for some niche use cases. I have pushed an experimental image on the Docker repository that might resolve your issue under the tag `0.3.1x`; please try this an see if it resolves your issue.

Thanks for your patience.

**Comment by jaid33** [comment]
Seems like everything works just fine with 0.3.1x 
Thank you very much for your effort!

Really strange with the 0.2.0 architecture thing... the 6 month were just guessed, but I used for sure longer than 3 weeks and didn't build anything.

**Comment by concision** [comment]
Glad to hear it works, this project is a mess of hacks to get working - it is a miracle anything about it works. I have released `0.3.1` along with the `latest` tag on the Docker repository. I recommend moving over to one of these tags asap, as I will likely deprecate the `0.3.1x` tag soon. I am not sure how it could have worked before, but regardless, thank you for filing an issue about issues with the armv7 support.

I think part of the issue might have been an oversight of mine on which ubuntu repository I was pulling from.

I will be closing this issue as it appears to be resolved. If you think this is a mistake or experience further issues with the `linux/arm/v7` build, you can reopen this issue or create a new issue.

**Comment by jaid33** [comment]
I think something went wrong, 0.3.1 is not working and the image size is just like 0.3.0. 
Did you mix something up?

**Comment by concision** [comment]
Ugh. My apologies, it appears I have made a sloppy mistake. It should be fixed now.

edit: it appears that my build was fired _before_ merging my branch for the arm fix, and not after

**Comment by jaid33** [comment]
Yep, looks good!

**Comment by concision** [comment]
Perfect, thank you for your patience and testing help!
